src/item.cpp

```diff
@@ -4697,8 +4696,10 @@ item::reload_option::reload_option( const player *who, const item *target, const
         ? this->ammo->contents.front()
         : *this->ammo;
 
-    if( this->ammo->is_watertight_container() || ( tmp.is_ammo() && !target->has_flag( "RELOAD_ONE" ) ) ) {
-        qty( tmp.charges );
+    if( this->ammo->is_watertight_container() ||
+        ( tmp.is_ammo() && !target->has_flag( "RELOAD_ONE" ) ) ||
+        ( tmp.is_magazine() && tmp.has_flag( "SPEEDLOADER" ) ) ) {
+        qty( tmp.ammo_remaining() );
```

Comment:
Magazines work without this and I don't see why would speedloaders need to be special cased here. If they do need this check, there is probably an error somewhere else.

Reply:
This isn't the magazine code path, this is the integral_magazine codepath.